tf-modules/cumulus/internal-lambdas.tf

```diff
@@ -0,0 +1,18 @@
+module "sqs_message_remover_lambda" {
+  source = "../sqs-message-remover-lambda"
+
+  prefix = var.prefix
+
+  cmr_environment = var.cmr_environment
+
+  system_bucket = var.system_bucket
+
+  lambda_subnet_ids = var.lambda_subnet_ids
+  security_group_ids = var.security_group_ids
+
+  tags = var.tags
+
+  # is this necessary or should we move towards least privileges
```

Comment:
This probably shouldn't be merged in the final PR, let's have that discussion here / ticket it.     Sorting our mega-role is a definite thing that needs addressed.

Reply:
It's pretty low effort to implement the perms/role for this lambda, so I'm inclined to "just do it" in this case. it just needs to receive the resources as params that it should have access to

Reply:
That seems fine, here.    My only concern with the approach (generally, not specific to this PR) is that we take a consistent approach, and avoid consequences where we change the processing role assuming all lambdas use it, and have odd failures/etc when something isn't updated.

Reply:
I'm going to punt on this for now and it's certainly an improvement we can (and should) make later